# Assignment 3

### Instructions
1. Import the following modules:
    - `pandas`
    - `stats` from `scipy`
2. Create a dataframe from the data contained in *income_data.csv* using `pandas`

In [2]:
# Code here
import pandas as pd
from scipy import stats
df = pd.read_csv("Income Data.csv")

3. Calculate and display Pearson's **correlation coefficients** and the **p-values** for *income* and *education*

In [3]:
# Code here
correlation, p_value = stats.pearsonr(df["income"], df["education"])

print("Pearson Correlation Coefficient:", correlation)
print("P-value:", p_value)

Pearson Correlation Coefficient: 0.7245123564790015
P-value: 1.839946418520884e-08


3. Calculate and display Pearson's **correlation coefficients** and the **p-values** for *income* and *prestige*

In [4]:
# Code here
correlation, p_value = stats.pearsonr(df["income"], df["prestige"])

print("Pearson Correlation Coefficient:", correlation)
print("P-value:", p_value)

Pearson Correlation Coefficient: 0.8378013725507842
P-value: 7.144355594290168e-13


4. Calculate and display Pearson's **correlation coefficients** and the **p-values** for *education* and *prestige*

In [5]:
# Code here
correlation, p_value = stats.pearsonr(df["education"], df["prestige"])

print("Pearson Correlation Coefficient:", correlation)
print("P-value:", p_value)

Pearson Correlation Coefficient: 0.8519156284206675
P-value: 1.1708786264586817e-13


5. In the empty `Markdown` cell below, explain the following for **each of the calculated correlation coefficients**
    - The **strength** and **direction** of the correlation
    - The **statistical significance** of the test (**p-value**); emphasize if we can trust the correlation test or not, and why

## 5
## Income and Education
The correlation coefficients for income and education has a medium  correlation of 0.72. The direction is positive. This test can be trusted because it has a p value of less than 0.05 and once the P-Value of 1.839946418520884e-08 is converted to decimal of 0.0000000184 is way smaller than 0.05 meaning that this test is very trustworthy.

## Income and Prestige
The correlation coefficient of 0.83 is a very strong correlation. The direction of this is positive so as their income increases so does their prestige. This test is considered very trustworthy because the p-value of 7.144355594290168e-13 is equal to 0.000000000000714 which is way smaller than 0.05 meaning that it can be trusted.

## Education and Prestige
The correlation coefficient is very strong at 0.85. The direction of the correlation is positive which means that education and prestige increase together. This test is very trustworthy because the p-value of 1.1708786264586817e-13 is equal to 0.000000000000117 when converted to decimal, which again is way below the 0.05 value making it very dependable.

In the empty `Markdown` cell below, answer the following question:

6. Pearson's correlation coefficient OR covariance can both be used to measure the direction of the relationship between two variables (e.g., education and income). Explain why Pearson's correlation coefficient may be better than covariance.

## 6 
The reason Pearson's correlation coefficient is better than covariance is because of its fixed scale of -1.0 to +1.0. This is better because Pearson's correlation coefficient can tell you both the direction and strength of the correlation because of the standardization of the range for the correlation coefficient. For example a negative number leading to a negative direction, and positive number leading to a positive direction of growth. Also with Pearson's correlation coefficient you can test the reliability of the test based on the p-value. However covariance only gives the direction, but not the strength, or the p-value/statistical significance. The reason you cannot determine this reliably from covariance is because it is very dependent on the range of the data. If you change the range from a smaller to bigger range the covariance will change. Pearson's correlation fixes this by dividing the covariance by the standard deviation which gets rid of the issue with different ranges/units.

7. Submit this `Jupyter` file to D2L, renamed as **Last_First_Assignment3.ipynb** 
    - Replace '**Last**' and '**First**' with your name